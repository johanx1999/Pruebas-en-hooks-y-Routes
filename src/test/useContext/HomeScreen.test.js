import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../components/useContext/HomeScreen';
import { UserContext } from '../../components/useContext/UserContext';

/* 
    Pruebas en el useContext
    -Debemos proveer lo que el componente HomeScreen necesita que es un user con doble llave porque 
    estamos regresando un objeto
    -Utilizamos mount porque shallow solo nos muestra en en el snapshot la informacion que esperamos
    que tenga una etiqueta pre que tenga HomeScreen en h1 (NOS RENDERIZA TODO EL COMPONENTE HIJO)

*/

describe('pruebas en el <HomeScreen />', () => {
    const user = {
        name: 'johan',
        email: 'johan@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>

            <HomeScreen/>

        </UserContext.Provider>
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    
    
})



