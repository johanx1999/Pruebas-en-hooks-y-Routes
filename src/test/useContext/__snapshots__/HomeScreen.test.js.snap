// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "johan",
   "email": "johan@gmail.com"
}
  </pre>
</div>
`;
